### ISSUE TITLE
Case test_encap_with_mirror_session of test_ipinip.py failed due to swss and other dockers could not start after invoking fixture setup_mirror_session

### ISSUE THREAD

**Issue by echuawu** [sonic-net/sonic-mgmt#11744]
**Description**
After running the full cases of tests/test_pretest.py, all the container autorestart functions would be disabled due to case test_disable_container_autorestart.
Then running tests/dualtor/test_ipinip.py, case **test_encap_with_mirror_session** would failed for sure. Case **test_encap_with_mirror_session** would invoke fixture **setup_mirror_session**, and it would invoke fixture **setup_uplink**. There are steps in fixture **setup_uplink**, which could restart teamd docker, it would lead swss and syncd docker exits. 
```
        # Unmask the service
        rand_selected_dut.shell_cmds(cmds="systemctl unmask teamd")
        # Restart teamd
        rand_selected_dut.shell_cmds(cmds="systemctl restart teamd")
```
Under the context of container autorestart functions disabled, so the affected dockers would not be up.

**Steps to reproduce the issue:**
1. Run case **test_disable_container_autorestart** of tests/test_pretest.py
2. Run case **test_encap_with_mirror_session** of tests/dualtor/test_ipinip.py

**Describe the results you received:**
Case **test_encap_with_mirror_session** would failed due to:

`Failed: PortChannel didn't startup`

**Describe the results you expected:**
Case pass even at the situation of docker autorestart function disabled

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by echuawu** [comment]
Checked with MSFT team, they said they have not observed any test_ipinip.py failures. Technically speaking, if case test_disable_container_autorestart of tests/test_pretest.py run, then run case test_encap_with_mirror_session of tests/dualtor/test_ipinip.py in the same regression, then it would failed.

**Comment by bingwang-ms** [comment]
@echuawu Which container can't startup? Is it teamd or swss/syncd?

**Comment by echuawu** [comment]
> @echuawu Which container can't startup? Is it teamd or swss/syncd?

The reason why LAGs didn't startup - **swss/syncd** dockers are down after executing of  **systemctl restart teamd** 

These docker are down because there are  "auto_restart": "disabled" for FEATURE in config_db:

        "teamd": {
            "auto_restart": "disabled",
       "syncd": {
            "auto_restart": "disabled",

        "swss": {
            "auto_restart": "disabled",

**Comment by bingwang-ms** [comment]
Hi @echuawu, I checked the test record internally but didn't see the same error. I believe that's because the `autorestart` is enabled somehow by some test cases. But the issue makes sense to me. I raised a PR to fix it https://github.com/sonic-net/sonic-mgmt/pull/12909. Can you please help review? Thanks

**Comment by echuawu** [comment]
> Hi @echuawu, I checked the test record internally but didn't see the same error. I believe that's because the `autorestart` is enabled somehow by some test cases. But the issue makes sense to me. I raised a PR to fix it #12909. Can you please help review? Thanks

Thank you for debugging and updating! Sure, reviewed.
